.devcontainer/postCreateCommand.sh

```diff
@@ -1,5 +1,5 @@
 #! /usr/bin/env bash
-
-python -m pip install -r requirements-dev.txt
-python -m pip install -e .
-pre-commit install --install-hooks
+python -m pip install poetry
+poetry config virtualenvs.in-project true --local
```

Comment:
I agree that it would be good to put a more complete `.gitignore` into place. I also think it makes sense to do that in the context of this PR where we are changing substantial aspects of the build, & development workflow. I've added the additional content for the Python template.